Examples/KerrBH/KerrBHLevel.cpp

```diff
@@ -84,3 +87,26 @@ void KerrBHLevel::computeTaggingCriterion(FArrayBox &tagging_criterion,
 {
     BoxLoops::loop(ChiTaggingCriterion(m_dx), current_state, tagging_criterion);
 }
+
+void KerrBHLevel::specificPostTimeStep()
+{
+    CH_TIME("KerrBHLevel::specificPostTimeStep");
+    if (m_p.activate_extraction == 1)
+    {
+        // Populate the Weyl Scalar values on the grid
+        fillAllGhosts();
```

Comment:
Comment should say Madm not Weyl. 

I have always thought that filling the Weyl/Madm data on every level post timestep is quite inefficient. This is especially true here when we only actually use it every coarse or second to coarse timestep. 

Perhaps we could do something like
if(m_time % extraction_dt == 0) {calculate diagnostics}
I think the modulo thing is usually non trivial as you have to check to numerical precision. Having a small function in GRAMRLevel that returns true if m_time is a multiple of dt for a given level would be quite useful generally, and could then be used here. 

This can be filed as an issue if you don't want to do it now :-)

Reply:
A quick alternative is to just move it to inside the following `if(m_level == m_p.extraction_params.min_extraction_level())` no? Like that it is only calculated on the moments we want output. I'm not sure this is the same as what you wanted, let me know.

Reply:
Careful @TaigoFr, you want the `BoxLoops::loop` called on all levels, not just the extraction level.

Reply:
I realized that and edited my comment. I solved it in the last commit. It's a solution that is stable under numerical perturbations.

Reply:
Nice fix :-)
Can we also add it to BinaryBHLevel::specificPostTimeStep()? It probably doesn't save us much, but it annoys me that we calculate Weyl on every level timestep.